tests/functional/adapter/concurrent_transaction/test_concurrent_transaction.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+import threading
+
+from dbt.tests.util import run_dbt
+from dbt.tests.adapter.simple_seed.test_seed import SeedConfigBase
+from tests.functional.adapter.concurrent_transaction.fixtures import *
+
+
+class BaseConcurrentTransaction(SeedConfigBase):
+    def __init__(self):
+        self.query_state = None
+
+    def reset(self):
+        self.query_state = {
+            'view_model': 'wait',
+            'model_1': 'wait',
+        }
+
+    @pytest.fixture(scope="class")
+    def setUp(self, project):
+        self.reset()
+
+    @pytest.fixture(scope="class")
+    def schema(self):
+        return "concurrent_transaction"
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {
+            "udfs.sql": create_udfs_sql
+        }
+
+    def run_select_and_check(self, rel, sql, project):
+        connection_name = '__test_{}'.format(id(threading.current_thread()))
+        try:
+            with project.adapter.connection_named(connection_name):
+                client = project.adapter.connections \
+                    .get_thread_connection().handle
+                res = project.run_sql(sql, 'one', client)
+
+                if res[0]:
+                    self.query_state[rel] = 'good'
+                else:
+                    self.query_state[rel] = 'bad'
+
+        except Exception as e:
```

Comment:
Generally catching all Exceptions is bad practice. Can we catch the one that we expect? Better than that, if we expect an error, can we assert that we get that error in that scenario? Otherwise there should be no exception catch in the test case. Also, the if block is not needed. The two conditions result in the same code.